extensions/PhabBugz/lib/Feed.pm

```diff
@@ -358,6 +358,79 @@ sub _is_uplift_request_form_change {
   return $story_text =~ /\s+uplift request field/;
 }
 
+sub process_uplift_request_form_change {
+  # Process an uplift request form change for the passed revision object.
+  my ($revision) = @_;
+
+  # Take no action if the form is empty.
+  # TODO test
+  if ($revision->uplift_form->is_empty) {
+    INFO("Uplift request form field cleared, ignoring.");
+    return;
+  }
+
+  my $revision_phid = $revision->phid;
+  INFO(
+    "Uplift request form submitted on $revision_phid, " .
+    "requesting `#release-managers` review."
+  );
+
+  # Get `#release-managers` review group.
+  my $release_managers_group = Bugzilla::Extension::PhabBugz::Project
+    ->new_from_query({name => 'release-managers'});
+
+  if (!$release_managers_group) {
+    WARN(
+      "Uplift request change detected but `#release-managers` was " .
+      "not found on Phabricator."
+    );
+    return;
+  }
+
+  my $release_managers_phid = $release_managers_group->phid;
+
+  # Request `#release-managers` review on each revision.
+  foreach my $stack_revision_phid (keys %{$revision->stack_graph_raw}) {
+    # Query Phabricator for each revision object related to the updated revision.
+    my $stack_revision = Bugzilla::Extension::PhabBugz::Revision->new_from_query(
+      {phids => [$stack_revision_phid]}
+    );
+
+    # Add `#release-managers!` review and set revision status.
+    $stack_revision->add_reviewer("blocking($release_managers_phid)");
+    $stack_revision->update();
+
+    INFO("Requested #release-managers review of $stack_revision_phid.");
+  }
+
+  # Comment the uplift form rendered on Bugzilla
+  # TODO add rendered form to storage on Phabricator side.
+  # TODO test
+  my $comment_data = {
+    'bug_id' => $revision->bug->id,
+    'is_markdown' => 1,
+  };
+  my $comment = Bugzilla::Comment->insert_create_data($comment_data);
+
+  # If manual QE is required, set the Bugzilla flag
+  # TODO test
+  if ($revision->uplift_form->{"Needs manual QE test"}) {
+    # TODO how to set bug flags?
+    # TODO set the `qe-verify` flag to `?` here.
+    my $new_flags = {
+      type_id  => $flagtype->id,
+      status   => '?',
+      setter   => $setter,
```

Comment:
$setter will need to be the ID of the user setting the flag. Would this be phab-bot?

Reply:
Yes in this case it would be `phab-bot`. How would I get the ID for that user?

We could also potentially use the `$changer` variable from `process_revision_change`. Can we assume that any user who has access to a revision (ie to submit the uplift form on Phabricator) would also be able to update the bug as needed? If that's the case I'd be fine with making all these bug changes as the user who submitted the change in Phabricator.

Reply:
> Yes in this case it would be `phab-bot`. How would I get the ID for that user?
> 
> We could also potentially use the `$changer` variable from `process_revision_change`. Can we assume that any user who has access to a revision (ie to submit the uplift form on Phabricator) would also be able to update the bug as needed? If that's the case I'd be fine with making all these bug changes as the user who submitted the change in Phabricator.

`my $phab_user = Bugzilla::User->new({name => PHAB_AUTOMATION_USER});`

It might be that the $changer does not have access to the bug and so it is better that we use the phab-bot user making the bug changes. 

In process_revision_change(), we set the current user the phab-bot user already so it is making any changes to bugs. 
It happens at: 

`my $restore_prev_user = set_phab_user();`